### ISSUE TITLE
net::ERR_CONNECTION_REFUSED at the end of language pack download process

### ISSUE THREAD

**Issue by radinamatic** [learningequality/ka-lite#4628]
`net::ERR_CONNECTION_REFUSED` at the end of language pack download process

**Branch**: develop  (0.16)

**Current Behavior**: I've been getting this error at the very end of language pack download process:

![finished-langpack-da2](https://cloud.githubusercontent.com/assets/1457929/10491286/3221bae2-72a6-11e5-80fd-cdb79ee9d426.png)

If the page is refreshed, all 4 language packs that I downloaded previously appear correctly and are functional, but that screen just gives bad usage experience. 

I used the MINOR_VERSION "hack" by @MCGallaspy to get the 0.15 language packs into develop, so I'm not sure if that might be the problem.

**Comment by aronasorman** [comment]
This can only mean one thing: a server is down somewhere.

**Comment by rtibbles** [comment]
Check your terminal output, @radinamatic - I imagine the server died after the language pack downloaded. This is still very much a WIP.

**Comment by radinamatic** [comment]
The error repeated 4 times on exact same point on 4 different langpacks...

I've saved the terminal output from last night, just have to remember where exactly... :stuck_out_tongue:

**Comment by aronasorman** [comment]
Thanks @radinamatic! Language packs will probably stop working for a while as @rtibbles and co. are continuing the refactoring.

**Comment by radinamatic** [comment]
I'll try to attach the full terminal output, but it's long... Here is the gist: it's as if server autostarts after completing langpack download:

```
...
DEBUG:kalite:Found language pack en
DEBUG:kalite:Found language pack es
[14/Oct/2015 03:07:22] "GET /api/languagepacks/installed?_=1444784812745 HTTP/1.1" 200 372
[14/Oct/2015 03:07:22] "GET /static/fonts/glyphicons-halflings-regular.woff2 HTTP/1.1" 304 0
[14/Oct/2015 03:07:22] "GET /static/images/jquery-ui-icons/ui-bg_flat_75_ffffff_40x100.png HTTP/1.1" 304 0
[14/Oct/2015 03:07:22] "GET /securesync/api/info?fields=online&_=1444784812746 HTTP/1.1" 200 54
[14/Oct/2015 03:07:22] "GET /api/updates/progress?process_name=languagepackdownload&_=1444784812747 HTTP/1.1" 200 2
[14/Oct/2015 03:07:22] "GET /api/updates/progress?process_name=languagepackdownload&_=1444784812748 HTTP/1.1" 200 2
[14/Oct/2015 03:07:23] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812749 HTTP/1.1" 200 2
[14/Oct/2015 03:07:24] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812750 HTTP/1.1" 200 2
[14/Oct/2015 03:07:25] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812751 HTTP/1.1" 200 2
[14/Oct/2015 03:07:26] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812752 HTTP/1.1" 200 2
[14/Oct/2015 03:07:28] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812753 HTTP/1.1" 200 2
[14/Oct/2015 03:07:28] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812754 HTTP/1.1" 200 2
[14/Oct/2015 03:07:29] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812755 HTTP/1.1" 200 2
[14/Oct/2015 03:07:30] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812756 HTTP/1.1" 200 2
[14/Oct/2015 03:07:31] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812757 HTTP/1.1" 200 2
[14/Oct/2015 03:07:32] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812758 HTTP/1.1" 200 2
[14/Oct/2015 03:07:33] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812759 HTTP/1.1" 200 2
[14/Oct/2015 03:07:34] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812760 HTTP/1.1" 200 2
[14/Oct/2015 03:07:35] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812761 HTTP/1.1" 200 2
[14/Oct/2015 03:07:36] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812762 HTTP/1.1" 200 2
[14/Oct/2015 03:07:37] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812763 HTTP/1.1" 200 2
[14/Oct/2015 03:07:38] "GET /api/updates/progress?process_name=languagepackdownload&start_time=2015-10-14T01:07:22.000Z&_=1444784812764 HTTP/1.1" 200 2
[14/Oct/2015 03:07:39] "POST /api/languagepacks/start HTTP/1.1" 200 71
INFO:kalite:Downloading language pack for lang_name=Deutsch, software_version=0.15
DEBUG:kalite:Setting languagepackdownload to 6 total stages.
INFO:kalite:Downloading language pack 'de'
INFO:kalite:Retrieving language pack: de
DEBUG:kalite:Downloading zip from http://staging.learningequality.org/media/language_packs/0.15/de.zip
INFO:kalite:Unpacking language pack 'de'
INFO:kalite:Unpacking new translations
INFO:kalite:Creating static files for language pack 'de'
WARNING:kalite:No de_icu.js file found in locale_path /home/didi/ka-lite/locale/
INFO:kalite:Writing i18nized js file to /home/didi/ka-lite/content/locale/js/i18n/de.js
INFO:kalite:Moving files to their appropriate local disk locations.
DEBUG:kalite:Moving dubbed video map to /home/didi/ka-lite/locale/dubbed_video_mappings.json
DEBUG:kalite:Removing empty directory
WARNING:kalite:Could not find downloaded exercises; skipping: /home/didi/ka-lite/locale/de/exercises
INFO:kalite:Moving 1771 subtitles from /home/didi/ka-lite/locale/de/subtitles to /home/didi/ka-lite/kalite/static/srt/de/subtitles
INFO:kalite:Removing empty source directory (/home/didi/ka-lite/locale/de/subtitles).
INFO:kalite:Finished processing language pack Deutsch.
DEBUG:kalite:Completing process languagepackdownload
/home/didi/ka-lite/kalite/settings/base.py:43: RemovedInKALite_v016_Warning: We will be deprecating the old way of statically importing custom settings, in favor of a more flexible way. The easiest way to update your installation is to rename your local_settings.py (keeping it in the same directory) and add an import statement in the very first line of the new file so it looks like this:

    from kalite.project.settings.base import *
    # Put custom settings here...
    FOO = BAR

and then call kalite start with an additional argument pointing to your new settings module:

    kalite start --settings=kalite.my_settings

In the future, it is recommended not to keep your own settings module in the kalite code base but to put the file somewhere else in your python path, for instance in the current directory when running 'kalite --settings=my_module'.
  RemovedInKALite_v016_Warning
Validating models...

0 errors found
October 14, 2015 - 03:11:28
Django version 1.5.1, using settings 'kalite.project.settings.dev'
Development server is running at http://0.0.0.0:8008/
Quit the server with CONTROL-C.
DEBUG:kalite:Found language pack en
DEBUG:kalite:Found language pack es
DEBUG:kalite:Found language pack de
DEBUG:kalite:setting request language to en (session language en), from None
[14/Oct/2015 03:13:21] "GET /update/languages/ HTTP/1.1" 302 0
[14/Oct/2015 03:13:21] "GET /?login=True&next=%2Fupdate%2Flanguages%2F HTTP/1.1" 200 2242
[14/Oct/2015 03:13:22] "GET /_generated/dynamic.css HTTP/1.1" 200 63
[14/Oct/2015 03:13:22] "GET /_generated/dynamic.js HTTP/1.1" 200 373
[14/Oct/2015 03:13:22] "GET /static/js/i18n/en.js HTTP/1.1" 200 5476
[14/Oct/2015 03:13:22] "GET /static/fonts/style.css HTTP/1.1" 200 1502
[14/Oct/2015 03:13:22] "GET /jsreverse/ HTTP/1.1" 200 4395
[14/Oct/2015 03:13:22] "GET /data/khan/images/horizontal-logo-small.png HTTP/1.1" 200 4103
[14/Oct/2015 03:13:22] "GET /static/js/distributed/bundles/bundle_homepage.js HTTP/1.1" 200 76758
[14/Oct/2015 03:13:23] "GET /static/js/tota11y.min.js HTTP/1.1" 200 247599
[14/Oct/2015 03:13:23] "GET /data/khan/images/logo_10_enlarged_2.png HTTP/1.1" 200 119689
[14/Oct/2015 03:13:23] "GET /static/images/distributed/FLE.png HTTP/1.1" 200 2216
[14/Oct/2015 03:13:23] "GET /static/images/distributed/CC-BY-NC-SA.png HTTP/1.1" 200 697
[14/Oct/2015 03:13:24] "GET /static/js/distributed/bundles/bundle_base.js HTTP/1.1" 200 1050844
[14/Oct/2015 03:13:26] "GET /static/js/distributed/bundles/bundle_common.js HTTP/1.1" 200 4602786
[14/Oct/2015 03:13:27] "GET /securesync/api/user/status/?_=1444785178139 HTTP/1.1" 200 291
[14/Oct/2015 03:13:27] "GET /static/images/jquery-ui-icons/ui-bg_flat_75_ffffff_40x100.png HTTP/1.1" 200 178
[14/Oct/2015 03:13:27] "GET /static/fonts/glyphicons-halflings-regular.woff2 HTTP/1.1" 200 18028
[14/Oct/2015 03:13:33] "POST /securesync/api/user/login/ HTTP/1.1" 200 50
DEBUG:kalite:setting request language to en (session language en), from None
DEBUG:kalite:setting request language to en (session language en), from None
[14/Oct/2015 03:13:33] "GET /securesync/api/user/status/?_=1444785178140 HTTP/1.1" 200 297
DEBUG:kalite:Found language pack en
DEBUG:kalite:Found language pack es
DEBUG:kalite:Found language pack de
[14/Oct/2015 03:13:34] "GET /update/languages/ HTTP/1.1" 200 3403
...
```

**Comment by radinamatic** [comment]
Hmm, attaching anything a part from img files requires write permission to this repository...

**Comment by rtibbles** [comment]
@radinamatic please confirm fix of this during 0.16 testing.

**Comment by radinamatic** [comment]
Non extant on latest installers. Closing as fixed!
